applications/StructuralMechanicsApplication/custom_response_functions/adjoint_schemes/adjoint_structural_scheme.h

```diff
@@ -0,0 +1,422 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Martin Fusseder, https://github.com/MFusseder 
+//   
+
+#if !defined(KRATOS_ADJOINT_STRUCTURAL_SCHEME)
+#define KRATOS_ADJOINT_STRUCTURAL_SCHEME
+
+// System includes
+#include <vector>
+#include <string>
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "includes/element.h"
+#include "includes/condition.h"
+#include "includes/communicator.h"
+#include "includes/model_part.h"
+#include "includes/process_info.h"
+#include "includes/kratos_parameters.h"
+#include "includes/ublas_interface.h"
+#include "utilities/openmp_utils.h"
+#include "solving_strategies/schemes/scheme.h"
+#include "containers/variable.h"
+
+// Application includes
+#include "custom_response_functions/response_utilities/adjoint_structural_response_function.h"
+
+namespace Kratos
+{
+///@addtogroup StructuralMechanicsApplication
+///@{
+
+///@name Kratos Classes
+///@{
+
+/// A scheme for for adjoint equations.
+/**
+ *
+ *
+ */
+template <class TSparseSpace, class TDenseSpace>
+class AdjointStructuralScheme : public Scheme<TSparseSpace, TDenseSpace>
```

Comment:
does this work for static & dynamic problems?
If not then this should be included in the ClassName, see the one in the core for regular static problems